oracle_cardano/processor/tx_processors/success/bridging_requested_processor_skyline.go

```diff
@@ -0,0 +1,355 @@
+	"github.com/Ethernal-Tech/apex-bridge/oracle_cardano/core"
+	"github.com/Ethernal-Tech/apex-bridge/oracle_cardano/utils"
+	cCore "github.com/Ethernal-Tech/apex-bridge/oracle_common/core"
+	cUtils "github.com/Ethernal-Tech/apex-bridge/oracle_common/utils"
+	"github.com/Ethernal-Tech/cardano-infrastructure/indexer"
+	"github.com/hashicorp/go-hclog"
+)
+
+var _ core.CardanoTxSuccessProcessor = (*BridgingRequestedProcessorSkylineImpl)(nil)
+
+type BridgingRequestedProcessorSkylineImpl struct {
+	logger hclog.Logger
+}
+
+func NewNativeBridgingRequestedProcessor(logger hclog.Logger) *BridgingRequestedProcessorSkylineImpl {
+	return &BridgingRequestedProcessorSkylineImpl{
+		logger: logger.Named("bridging_requested_processor_skyline"),
+	}
+}
+
+func (*BridgingRequestedProcessorSkylineImpl) GetType() common.BridgingTxType {
+	return common.BridgingTxTypeBridgingRequest
+}
+
+func (*BridgingRequestedProcessorSkylineImpl) PreValidate(tx *core.CardanoTx, appConfig *cCore.AppConfig) error {
+	return nil
+}
+
+func (p *BridgingRequestedProcessorSkylineImpl) ValidateAndAddClaim(
+	claims *cCore.BridgeClaims, tx *core.CardanoTx, appConfig *cCore.AppConfig,
+) error {
+	metadata, err := common.UnmarshalMetadata[common.BridgingRequestMetadata](common.MetadataEncodingTypeCbor, tx.Metadata)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal metadata: tx: %v, err: %w", tx, err)
+	}
+
+	if metadata.BridgingTxType != p.GetType() {
+		return fmt.Errorf("ValidateAndAddClaim called for irrelevant tx: %v", tx)
+	}
+
+	p.logger.Debug("Validating relevant tx", "txHash", tx.Hash, "metadata", metadata)
+
+	err = p.validate(tx, metadata, appConfig)
+	if err == nil {
+		p.addBridgingRequestClaim(claims, tx, metadata, appConfig)
+	} else {
+		//nolint:godox
+		// TODO: Refund
+		// p.addRefundRequestClaim(claims, tx, metadata)
+		return fmt.Errorf("validation failed for tx: %s, err: %w", tx.Hash, err)
+	}
+
+	return nil
+}
+
+func (p *BridgingRequestedProcessorSkylineImpl) addBridgingRequestClaim(
+	claims *cCore.BridgeClaims, tx *core.CardanoTx,
+	metadata *common.BridgingRequestMetadata, appConfig *cCore.AppConfig,
+) {
+	totalAmountSrc := uint64(0)
+	totalAmountDest := uint64(0)
+	totalAmountWrappedTokenSrc := uint64(0)
+	totalAmountWrappedTokenDest := uint64(0)
+
+	cardanoDestConfig := appConfig.CardanoChains[metadata.DestinationChainID]
+
+	receivers := make([]cCore.BridgingRequestReceiver, 0, len(metadata.Transactions))
+
+	for _, receiver := range metadata.Transactions {
+		receiverAddr := strings.Join(receiver.Address, "")
+
+		if receiver.Additional == nil {
+			receiver.Additional = &common.BridgingRequestMetadataCurrencyInfo{}
+		}
+
+		if receiverAddr == cardanoDestConfig.BridgingAddresses.FeeAddress {
+			// fee address will be added at the end
+			continue
+		}
+
+		var (
+			receiverAmount        uint64
+			receiverAmountWrapped uint64
+		)
+
+		if receiver.IsNativeTokenOnSrc {
+			receiverAmount = receiver.Amount + receiver.Additional.DestAmount
```

Comment:
please add comment on this line why we are adding ` receiver.Additional.DestAmount`